src/libraries/System.Net.Requests/src/System/Net/TaskExtensions.cs

```diff
@@ -18,20 +18,7 @@ internal static class TaskExtensions
 
             task.ContinueWith(completedTask =>
             {
-                bool shouldInvokeCallback = false;
-
-                if (completedTask.IsFaulted)
-                {
-                    shouldInvokeCallback = tcs.TrySetException(completedTask.Exception!.InnerExceptions);
-                }
-                else if (completedTask.IsCanceled)
-                {
-                    shouldInvokeCallback = tcs.TrySetCanceled();
-                }
-                else
-                {
-                    shouldInvokeCallback = tcs.TrySetResult(completedTask.Result);
-                }
```

Comment:
@pavelsavara, it looked like https://github.com/dotnet/runtime/blob/cc59b89c559036ca5a095818c7763df10283c6be/src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSWebWorker.cs#L175-L200 could also be replaced with a single call to TrySetFromTask, except I wasn't sure about that _bodyRes != null check and its implications.

Reply:
I wanted to share same code for both `Task` and `Task<T>`.
There are `_bodyVoid` and `_bodyRes` callbacks.
When `_bodyRes!=null` tells me that I have to propagate the result value.

And so my `_task` is sometimes `Task<T>` and sometimes void `Task`. But I have `TaskCompletionSource<T>` or `TaskCompletionSource<int>`, where generic parameter `int` is playing the role of dummy void.

Is there better way how to overcome difference between `TaskCompletionSource<T>` and `TaskCompletionSource` without duplicating many methods ? Could we have common interface on them ?